packages/pigeon/platform_tests/test_plugin/windows/pigeon/core_tests.gen.cpp

```diff
@@ -2578,15 +2761,8 @@ void HostIntegrationCoreApi::SetUp(flutter::BinaryMessenger* binary_messenger,
             try {
               const auto& args = std::get<EncodableList>(message);
               const auto& encodable_an_enum_arg = args.at(0);
-              const int64_t an_enum_arg_value =
-                  encodable_an_enum_arg.IsNull()
-                      ? 0
-                      : encodable_an_enum_arg.LongValue();
-              const auto an_enum_arg =
-                  encodable_an_enum_arg.IsNull()
-                      ? std::nullopt
-                      : std::make_optional<AnEnum>(
-                            static_cast<AnEnum>(an_enum_arg_value));
+              const auto* an_enum_arg = &(std::any_cast<const AnEnum&>(
+                  std::get<CustomEncodableValue>(encodable_an_enum_arg)));
```

Comment:
The bad variant access error is coming from this pattern: if the arg's value is null, then `encodable_an_enum_arg` isn't a `CustomEncodableValue` at all, it's a monotype (i.e., `IsNull()` is true). So you need to check `IsNull()` before doing a std::get instead of after, or you need to use a `get_if<CustomEncodableValue>(&encodable_an_enum_arg)` and then if that's null, pass null, and if it's not not, do the any-cast. (The thing that tripped me up in my first attempt to write a fix for this is that you can't `any_cast` a `CustomEncodableValue*` because of how implicit conversions work, so you'll need to do some gymnastics with a value variable like before I think.)

Reply:
This is kind of clunky, but the only way I can find to do it in a single statement involves a pointer to a reference that I can't prove to myself has a lifetime beyond that statement, and I'd rather the code be longer than have undefined behavior...

```cpp
AnEnum an_enum_arg_value;
const AnEnum* an_enum_arg = nullptr;
if (!encodable_an_enum_arg.IsNull()) {
  an_enum_arg_value = std::any_cast<AnEnum>(std::get<CustomEncodableValue>(encodable_an_enum_arg));
  an_enum_arg = &an_enum_arg_value;
}
...EchoNullableEnum(an_enum_arg);
```